doc-surrealdb_versioned_docs/version-1.x/reference-guide/vector-search.mdx

```diff
@@ -0,0 +1,219 @@
+
+```surql
+CREATE actor:1 SET name = 'Actor 1', embedding = [0.1, 0.2, 0.3, 0.4];
+CREATE actor:2 SET name = 'Actor 2', embedding = [0.2, 0.1, 0.4, 0.3];
+CREATE actor:3 SET name = 'Actor 3', embedding = [0.4, 0.3, 0.2, 0.1];
+CREATE actor:4 SET name = 'Actor 4', embedding = [0.3, 0.4, 0.1, 0.2];
+```
+
+Output:
+```surql
+[[{ embedding: [0.1f, 0.2f, 0.3f, 0.4f], id: actor:1, name: 'Actor 1' }], 
+[{ embedding: [0.2f, 0.1f, 0.4f, 0.3f], id: actor:2, name: 'Actor 2' }], 
+[{ embedding: [0.4f, 0.3f, 0.2f, 0.1f], id: actor:3, name: 'Actor 3' }], 
+[{ embedding: [0.3f, 0.4f, 0.1f, 0.2f], id: actor:4, name: 'Actor 4' }]]
+```
+
+You can assume your facial features to be
+
+```surql
+LET $person_embedding = [0.15, 0.25, 0.35, 0.45];
+```
+
+You have the option of using 3 different approaches for performing Vector search. 
+
+- [Bruteforce](/docs/surrealdb/surrealql/operators#brute-force-method-since-130)
+- [Hierarchical navigable small world(HNSW)](/docs/surrealdb/surrealql/statements/define/indexes#hnsw-hierarchical-navigable-small-world-since-150)
+- [M-Tree](/docs/surrealdb/surrealql/statements/define/indexes#m-tree-index-since-130)
+
+As you want to perform a nearest neighbour search and not an exact search, you would typically use an index like [HNSW](/docs/surrealdb/surrealql/statements/define/indexes#hnsw-hierarchical-navigable-small-world-since-150) or [M-Tree](/docs/surrealdb/surrealql/statements/define/indexes#m-tree-index-since-130). 
+
+```surql
+DEFINE INDEX mt_pts ON actor FIELDS embedding MTREE DIMENSION 4 TYPE F32;
+SELECT id, name, vector::similarity::cosine(embedding, $person_embedding) AS dist FROM actor WHERE embedding <|2|> $person_embedding;
+SELECT id, vector::distance::euclidean(embedding, $person_embedding) AS dist FROM actor WHERE embedding <|1|> $person_embedding;
+```
+
+Output:
+```surql
+[[{ dist: 0.9979654098963515f, id: actor:1, name: 'Actor 1' }, 
+{ dist: 0.9409388150451313f, id: actor:2, name: 'Actor 2' }]]
+```
+
+Now that you know how to draft a vector search query in SurrealQL, let's take a step back and understand some of its terms and concepts. 
+
+## How to store vector embeddings
+
+To store vectors in SurrealDB, you typically define a field within your data schema dedicated to holding the vector data. These vectors represent space data points and can be used for various applications, from recommendation systems to image recognition. Below is an example of how to create records with vector embeddings:
+
+```surql
+CREATE Document:1 CONTENT {
+  "items": [
+    {
+      "content": "apple",
+      "embedding": [0.00995, -0.02680, -0.01881, -0.08697]
+    }
+  ]
+};
+```
+
+The vector is represented as an array of floating-point numbers.
```

Comment:
```suggestion
The vector is represented as an array of numbers, commonly using floating-point values. However, depending on the model, it can also be an array of integers. The exact type can be defined in the index. 
```